include/monad/result.hpp

```diff
@@ -244,3 +245,37 @@ struct fmt::formatter<Result<T>> {
     }
   }
 };
+
+/**
+ * @brief Filter function for testing if a result has an error
+ *
+ * @param exp The expected type to test
+ * @tparam T The value type
+ * @tparam E The error type
+ * @return if the expected has an error
+ */
+template <typename T, typename E>
+constexpr bool has_error(const expected<T, E>& exp) {
+  return (!exp);
```

Comment:
Why the parens? I assume there's a reason

Reply:
no, I just did it out of habit, I can drop them.